catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FindChildrenStreamEndPlugin.java

```diff
@@ -0,0 +1,168 @@
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * Find the children of the parent metacard and process the children in batches. The children will be
+ * processed in the order in which they were originally created.
+ */
+public class FindChildrenStreamEndPlugin implements StreamEndPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FindChildrenStreamEndPlugin.class);
+
+    private static final int BATCH_SIZE = 50;
+
+    private final Factory factory;
+
+    public FindChildrenStreamEndPlugin(Factory factory) {
+        this.factory = factory;
+    }
+
+    public Factory getFactory() {
+        return factory;
+    }
+
+    @Override
+    public void streamEnded(Context context) {
+
+        synchronized (context.isParentDirty()) {
+
+            if (!context.isParentDirty()
+                    .get()) {
+                return;
+            }
+
+            context.getParentMetacard()
+                    .ifPresent(parentMetacard -> handleParentMetacard(context, parentMetacard));
+
+            context.isParentDirty()
+                    .set(false);
+        }
+    }
+
+    @Override
+    public void accept(Visitor visitor) {
+        visitor.visit(this);
+    }
+
+    private void handleParentMetacard(Context context, Metacard parentMetacard) {
+
+        Handler handler = factory.build();
+
+        try {
+
+            final FilterBuilder builder = new GeotoolsFilterBuilder();
+            Filter filter = builder.attribute(Associations.DERIVED)
+                    .is()
+                    .equalTo()
+                    .text(parentMetacard.getId());
+
+            int startIndex = 1;
+            int expectedReturnCount = 0;
+
+            do {
+                QueryRequest queryRequest = new QueryRequestImpl(new QueryImpl(filter,
+                        startIndex,
+                        BATCH_SIZE,
+                        new SortByImpl(Core.METACARD_CREATED, SortOrder.ASCENDING),
+                        true,
+                        0));
+
+                QueryResponse queryResponse = context.getUdpStreamProcessor()
+                        .getCatalogFramework()
+                        .query(queryRequest);
+
+                if (startIndex == 1) {
+                    expectedReturnCount = Math.toIntExact(queryResponse.getHits());
+                }
+
+                List<Metacard> batch = queryResponse.getResults()
+                        .stream()
+                        .map(Result::getMetacard)
+                        .collect(Collectors.toList());
+
+                handler.handle(context, parentMetacard, batch);
+
+                startIndex += batch.size();
+
+            } while (startIndex <= expectedReturnCount);
+
+            handler.end(context, parentMetacard);
+
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
```

Comment:
* Because the handlers are created at the start of this method, there shouldn't be any problem if a call to CatalogFramework throws an exception. 

* Are you suggesting that we catch RuntimeException? If so, why? 

* I'll update the javadoc to reflect the current behavior.